qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,385 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Hamiltonian simulation of tridiagonal Toeplitz symmetric matrices."""
+
+import numpy as np
+from scipy.sparse import diags
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister
+from qiskit.circuit.library import UGate, MCMTVChain
+
+from .linear_system_matrix import LinearSystemMatrix
+
+
+class Tridiagonal(LinearSystemMatrix):
+    """Class of tridiagonal Toeplitz symmetric matrices"""
+
+    def __init__(self, num_state_qubits: int, main_diag: float, off_diag: float,
+                 tolerance: float = 1e-2, evo_time: float = 1.0, trotter: int = 1,
+                 name: str = 'tridi') -> None:
+        """
+        Args:
+            num_state_qubits: the number of qubits where the unitary acts.
+            main_diag: the main diagonal entry
+            off_diag: the off diagonal entry
+            tolerance: the accuracy desired for the approximation
+            evo_time: the time of the Hamiltonian simulation
+            trotter: the number of Trotter steps
```

Comment:
Would this be better as `steps`, explaining that is the number of trotter steps, or call it `trotter_steps` where its more eslf evident. I guess we always trotterize and do not use any alternate

Reply:
changed to `trotter_steps` for clarity